great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -93,6 +93,16 @@
 try:
     import pybigquery.sqlalchemy_bigquery
 
+    ###
+    # NOTE: 20210816 - jdimatteo: A convention we rely on is for SqlAlchemy dialects
+    # to define an attribute "dialect". A PR has been submitted to fix this upstream
+    # with https://github.com/googleapis/python-bigquery-sqlalchemy/pull/251. If that
+    # fix isn't present, add this "dialect" attribute here:
+    if not hasattr(pybigquery.sqlalchemy_bigquery, "dialect"):
+        pybigquery.sqlalchemy_bigquery.dialect = (
+            pybigquery.sqlalchemy_bigquery.BigQueryDialect
+        )
+
```

Comment:
If this hasn't yet been added as a note in the Jira ticket, could you please do so? 😄 

Excited to see what the resolution is here. Let's also make an internal reminder to clean up this note upon the PR's resolution.

Reply:
Thank you :) It's GREAT-173